### ISSUE TITLE
Chocolatey config changes in 0.10.4+ - The process cannot access the file because it is being used by another process

### ISSUE THREAD

**Issue by ferventcoder** [chocolatey/choco#1241]
The fixes completed for #1047 are now causing the following error when using Puppet's Chocolatey provider to set configuration:

`The process cannot access the file because it is being used by another process`

This is likely due to the Puppet provider holding a lock on the original file. The original method renamed that file and wrote a new file in it's place, which allowed issues with something holding a lock on a file and so it was never caught in testing  - https://github.com/puppetlabs/puppetlabs-chocolatey/blob/5e3af2410863a216a4cb0dce90a74171f9873931/lib/puppet/provider/chocolateysource/windows.rb#L45-L48

https://tickets.puppetlabs.com/browse/MODULES-4678 is the issue for the Puppet provider.

~~This has been validated by folks using Puppet and outside of Puppet when they run Chocolatey. It's like something now has an lock on the file, except for the folks that have checked, there is nothing holding a lock on the file itself. Seems to be a race condition somewhere.~~

cc @RichiCoder1

~~~sh
C:\ProgramData\PuppetLabs>choco -version
This is try 1/3. Retrying after 300 milliseconds.
 Error converted to warning:
 (32) The process cannot access the file because it is being used by another process: [\\?\C:\ProgramData\chocolatey\config\chocolatey.config]
This is try 2/3. Retrying after 400 milliseconds.
 Error converted to warning:
 (32) The process cannot access the file because it is being used by another process: [\\?\C:\ProgramData\chocolatey\config\chocolatey.config]
Maximum tries of 3 reached. Throwing error.
Error deserializing response of type chocolatey.infrastructure.app.configuration.ConfigFileSettings:
 (32) The process cannot access the file because it is being used by another process: [\\?\C:\ProgramData\chocolatey\config\chocolatey.config]
(32) The process cannot access the file because it is being used by another process: [\\?\C:\ProgramData\chocolatey\config\chocolatey.config]
~~~

### More details

This appears to happen after:

~~~sh
Attempting to replace "C:\ProgramData\chocolatey\config\chocolatey.config"
 with "C:\ProgramData\chocolatey\config\chocolatey.config.update". Backup placed at "C:\ProgramData\chocolatey\config\chocolatey.config.backup".
~~~

### Workaround

Downgrade to Chocolatey v0.10.3 until this is resolved.

**Comment by ferventcoder** [comment]
If you run into this error, please provide the output of the following:

`cacls C:\ProgramData\chocolatey\config\chocolatey.config`

**Comment by tek0011** [comment]
On first run, and first failure it looks like this.  
![image](https://cloud.githubusercontent.com/assets/1085020/24719532/b99b4b26-19ff-11e7-8a4a-86db636fe68d.png)

On second run, same failure, same red wall of text, but this time it says it can find the files:
![image](https://cloud.githubusercontent.com/assets/1085020/24719567/cf301a98-19ff-11e7-94c8-a948abe4f701.png)

Second run:

```
C:\ProgramData\chocolatey\config>cacls C:\ProgramData\chocolatey\config\chocolatey.config
C:\ProgramData\chocolatey\config\chocolatey.config NT AUTHORITY\SYSTEM:(ID)F
                                                   BUILTIN\Administrators:(ID)F
                                                   BUILTIN\Users:(ID)R


C:\ProgramData\chocolatey\config>tree /f
Folder PATH listing for volume OSDisk SSD
Volume serial number is 9ED9-43A5
C:.
    chocolatey.config
    chocolatey.config.backup
    chocolatey.config.update

No subfolders exist
```

**Comment by tek0011** [comment]
[config.zip](https://github.com/chocolatey/choco/files/900349/config.zip)

configs at the time of issue, if helps

**Comment by ferventcoder** [comment]
Thanks! This will help

**Comment by sd4705** [comment]
Our site is affected by this problem. Thanks for working on this.

You can downgrade chocolatey with this command:
choco upgrade chocolatey --version 0.10.3 --force -y

Close the powershell window and then run puppet again:
puppet agent -t

**Comment by RichiCoder1** [comment]
This is very odd. File locks were something I was trying hard to avoid :|

**Comment by ferventcoder** [comment]
This change also affects DSC: 

> We started getting errors in our deployments recently and after doing a bit of digging here is what we found. First, Chocolatey was updated from 0.10.4 to 0.10.5 recently and with that change it seems the stick a backup of the chocolatey.config file in the config folder named chocolatey.config.backup right after install. This command in your test resource:  
> `$configfile = Get-ChildItem $configfolder | Where-Object {$_.Name -match "chocolatey.config"}`
> now returns two files and so the following line to parse the xml fails since there are two file objects return. I simply updated the -match to a -eq in the above line and that seems to fix the issue.

**Comment by ferventcoder** [comment]
I believe this is fixed for 0.10.6.

**Comment by ferventcoder** [comment]
Okay, now this is ready for 0.10.6.
